sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/NotificationHookModelTests.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.AI.MetricsAdvisor.Administration;
+using Azure.Core.TestFramework;
+using NUnit.Framework;
+
+namespace Azure.AI.MetricsAdvisor.Tests
+{
+    public class NotificationHookModelTests : MockClientTestBase
+    {
+        public NotificationHookModelTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        private string UnknownHookContent => $@"
+        {{
+            ""hookId"": ""{FakeGuid}"",
+            ""hookName"": ""unknownHookName"",
+            ""hookType"": ""unknownType"",
+            ""externalLink"": ""https://fakeuri.com/"",
+            ""description"": ""unknown hook description"",
+            ""admins"": [
+              ""foo@contoso.com""
+            ]
+        }}
+        ";
+
+        [Test]
+        public async Task NotificationHookGetsUnknownHook()
+        {
+            MockResponse getResponse = new MockResponse(200);
+            getResponse.SetContent(UnknownHookContent);
+
+            MetricsAdvisorAdministrationClient adminClient = CreateInstrumentedAdministrationClient(getResponse);
+            NotificationHook hook = await adminClient.GetHookAsync(FakeGuid);
+
+            Assert.That(hook.Id, Is.EqualTo(FakeGuid));
+            Assert.That(hook.Name, Is.EqualTo("unknownHookName"));
+            Assert.That(hook.ExternalUri.AbsoluteUri, Is.EqualTo("https://fakeuri.com/"));
+            Assert.That(hook.Description, Is.EqualTo("unknown hook description"));
+            Assert.That(hook.AdministratorEmails.Single(), Is.EqualTo("foo@contoso.com"));
+        }
+
+        [Test]
+        public async Task NotificationHookUpdatesUnknownHook()
```

Comment:
In this one we're making sure that, after getting back an unknown hook from the service, we can also update its common properties and send it back.